test/test.geometryutil.js

```diff
@@ -477,6 +477,53 @@ describe('Line order', function() {
   });
 });
 
+describe('Concatenate polylines', function(done) {
+
+  it('It should return an empty array if no polyline is given', function(done) {
+    var concated = L.GeometryUtil.concatLines(map, []);
+    assert.equal(concated.length, 0);
+    done();
+  });
+
+  it('It should return the original array if there is one given polyline', function(done) {
+    var line = L.polyline([[0, 0], [1, 1]]);
+    var concated = L.GeometryUtil.concatLines(map, [line]);
+    assert.equal(concated.length, 1);
+    done();
+  });
+
+  it('It should return an arry of concatenated lines', function(done) {
+    var lineA = L.polyline([[0, 0], [1, 1]]);
+    var lineB = L.polyline([[0.1, 0.1], [2, 1]]);
+    var lineC = L.polyline([[20, 20], [30, 30]]);
+    var lineD = L.polyline([[35, 35], [40, 40], [30, 30]]);
+    var concated = L.GeometryUtil.concatLines(map, [lineD, lineA, lineC, lineB], 1);
+
+    assert.equal(concated.length, 2);
+
+    var concatedLatLngs = concated[0].getLatLngs();
+    assert.equal(concatedLatLngs.length, 4);
+    assert.isTrue(concatedLatLngs[0].equals(L.latLng([35, 35])));
+    assert.isTrue(concatedLatLngs[3].equals(L.latLng([20, 20])));
+
+    var concatedLatLngs = concated[1].getLatLngs();
+    assert.equal(concatedLatLngs.length, 3);
+    assert.isTrue(concatedLatLngs[0].equals(L.latLng([2, 1])));
+    assert.isTrue(concatedLatLngs[2].equals(L.latLng([1, 1])));
+
+    done();
+  })
```

Comment:
I suggest that you add a couple of additional tests (simple ones with two lines for example), for the following cases:

* tolerance is 0
* the input paths are not modified
* the two paths form a «\/» or a /\»
* the two paths are equal
* the two paths are equal but opposite direction
* the two paths form a loop